### ISSUE TITLE
Roku client doesn't report as active device in dashboard

### ISSUE THREAD

**Issue by michaelkrieger** [jellyfin-archive/jellyfin-roku-legacy#778]
**Describe the bug**
A client on a roku 3 running Jellyfin-Roku 1.5.0 does not show a roku as an active device. The transcoded file is still transferring via http (as seen from traffic from the machine) but without seeing that there is no way to know the user is still streaming. 

**To Reproduce**
1. Stream movie from roku
2. Movie transcodes 
3. Client shows active and showing movie playing
4. Eventually client disappears from active client list and the list shows empty (or other users)

**Comment by neilsb** [comment]
Can I just clarify that the issue you're reporting is that the Roku device does not appear on the Web Dashboard at all?

![image](https://user-images.githubusercontent.com/5013482/188398326-8cd67aeb-e4af-48ba-8cd0-f518d1c49023.png)

Tests I've performed show it working correctly using the latest server (10.8.4).  Could you confirm what server you are using, and do you see other clients (web, android, etc) on the dashboard when they are streaming?

**Comment by michaelkrieger** [comment]
I’m using. 10.8.4 server.

it does appear, and does show the transcoding process.

after a while, maybe 10-20 minutes, the transcoded movie is still streaming to the client but that entry disappears

so it appears as you show it and at some point stops and there is no evidence of it. 

Your clarification is correct. I should have mentioned that it initially appears

**Comment by neilsb** [comment]
Do you happen to know in those cases if the client is still streaming?  While playing, the client updates the server every few seconds with the current status and playback position.  When the user ends playback (pressing stop, back, etc) then that is also communicated to the server.

However, with Roku, if the user exits the channel (e.g. Home button, etc) then Roku does not allow the currently running channel to gracefully shutdown and the process is simply terminated.  When this happens we have no chance to tell the server to stop the transcode stream.  In this case the server will eventually timeout the client and remove it from the dashboard.  However, while I would expect the server to continue transcoding for a while (and ffmpeg logs will show this happening) I would be surprised to continue to see http traffic to the client - as I would assume Roku would terminate the stream when it killed the channel.

If the client is still viewing the stream in the Jellyfin roku app however, then this is almost certainly a problem with the client.

I'll see if I can reproduce this.

**Comment by michaelkrieger** [comment]
I have had it disappear from the dashboard within the first 20 minutes and then continue to stream for over an hour, with the traffic going across the network.  I believe if there was no receiving application, it wouldn't be sending the traffic over the network.

I don't experience this with AndroidTV/FireTV clients, which terminate immediately. 
I'll see what information I can get next time I catch it.

**Comment by mdsib** [comment]
Does it work if you pause and unpause the video? If so, https://github.com/jellyfin/jellyfin-roku/pull/789 likely fixes this.

**Comment by alanazar** [comment]
> Does it work if you pause and unpause the video? If so, #789 likely fixes this.

I was able to replicate this issue while a video was transcoding & playing.
1st time it ran for ~8min and the 2nd time it ran for 7min when it disappeared from the Dashboard, even though the video was still transcoding (confirmed in task manager) and playing on the Roku.
Refreshing the dashboard didn't help or navigating away and coming back to it.
Pausing and playing the video made the dashboard pick it up again

I tested https://github.com/jellyfin/jellyfin-roku/pull/789 and the video played for 30min before I manually stopped it so it looks like it solved this issue.
